core-tests/src/test/java/org/mule/runtime/core/internal/connection/PoolingConnectionManagementStrategyTestCase.java

```diff
@@ -142,10 +155,60 @@ public void failDueToInvalidConnection() throws ConnectionException {
 
   @Test(expected = ConnectionException.class)
   public void failDueToNullConnectionValidationResult() throws ConnectionException {
+    initStrategy();
+    connection1 = strategy.getConnectionHandler();
+    connection2 = strategy.getConnectionHandler();
+
     when(connectionProvider.validate(anyVararg())).thenReturn(null);
     strategy.getConnectionHandler().getConnection();
   }
 
+  @Test
+  public void defaultInitializationPolicyIsHonored() throws ConnectionException {
```

Comment:
I would also assert how many elements are idle in the pool for all these new tests.

Reply:
You are right, dismiss this comment 😄